pkg/debug/scripts/collect-info.sh

```diff
@@ -234,6 +234,17 @@ collect_network_info()
     echo "- done network info"
 }
 
+collect_pillar_memory_backtraces()
+{
+    echo "- pillar memory backtraces"
+
+    eve http-debug > /dev/null 2>&1
+    eve exec debug curl --retry-all-errors --retry 5 -m 5 "http://127.1:6543/debug/pprof/heap?debug=1" > "$DIR/pillar-memory-backtraces" 2> /dev/null
```

Comment:
This is script is suppose to be executed only within debug container (where it's available)... you don't need `eve exec debug`... or is there other reason to do so?

Reply:
> This is script is suppose to be executed

Where do you have this information from? It starts from other containers, too.

Reply:
If somebody uses the normal console and plugs in the USB pen-drive with the script, they run it from outside the debug container.

Also I could imagine that we start it from pillar container in the future or f.e. via edgeview.

(Just because other parts of `collect-info.sh` might fail does not mean I should add more parts that fail, doesn't it?)

Reply:
if somebody plugs an USB stick and try to execute the script from the main console, it won't be able to use apk (read only file system) and most probably it will fail. The script was designed to run within debug container, as explained in the original commit: https://github.com/lf-edge/eve/commit/d9cc86c619

```
The script should be executed from the debug container, so:

   # eve enter debug (not needed if there is an SSH access)
   # collect-info.sh
```

This is the right way to call collect-info.sh. Call it from other containers might be error prone (we can't never guarantee full access by other containers). However, I agree this must better documented.

Reply:
`apk` might fail because there is no internet access, even in the debug container.

Just because other parts of collect-info.sh might fail does not mean I should add more parts that fail, doesn't it?

Reply:
> `apk` might fail because there is no internet access, even in the debug container.
> 
> Just because other parts of collect-info.sh might fail does not mean I should add more parts that fail, doesn't it?

following this logic you should add `exec debug` to all other commands that might not be available outside debug container: lspci, lsusb, lshw.... what I really wanted to know is why you added `exec debug`, it's because `curl` might not be available on other containers? Or there is another reason? If the reason it's because curl might not be available, then we have exactly the same situation for lshw, lsusb, lspci and it doesn't make sense to add `exec debug` only for curl.... if there is another reason, then it might justify use `exec debug`....

Reply:
ok, so as you answered, the reason to use `eve debug` it's because curl might not be available when running outside debug container... if you are really concerned about that, then the logic implemented in the script is to put in on PKG_DEPS (https://github.com/lf-edge/eve/blob/master/pkg/debug/scripts/collect-info.sh#L16) and enforce the package dependency, my point here is that it doesn't make sense to add `eve exec debug` only for curl, we have the same situation for all others tools, and it doesn't make sense to add `eve exec debug` for all of them (so no PRs to be done), the script was designed to run from debug container, where is expected to have lspci, lsusb, lshw, curl... another option would be use wget, which is implemented by busybox as well, so it should be widely available.... but let's not call another instance of debug container within debug container....

Reply:
> It is using the already existing one; see https://github.com/lf-edge/eve/blob/master/pkg/dom0-ztools/rootfs/bin/eve#L136 - it is not even forking!

ok, in this case I can accept the `eve exec debug`, although I reaffirm this will be useless most of time, and if eventually you face someone running it from another container or environment, curl will work, bust most probably many other tools will not be available, making the results far from complete....